src/enzo/New_Grid_ReadGrid.C

```diff
@@ -386,55 +387,58 @@ int grid::Group_ReadGrid(FILE *fptr, int GridID, HDF5_hid_t file_id,
 
     }
 
-
-    if (HydroMethod == MHD_RK) { // This is the MHD with Dedner divergence cleaning that needs an extra field
-      // 
-
-   
-      int activesize = 1;
-      for (int dim = 0; dim < GridRank; dim++)
-	activesize *= (GridDimension[dim]-2*NumberOfGhostZones);
-      
-      /* if we restart from a different solvers output without a Phi Field create here and set to zero */
-      int PhiNum; 
-      if ((PhiNum = FindField(PhiField, FieldType, NumberOfBaryonFields)) < 0) {
-	fprintf(stderr, "Starting with Dedner MHD method with no Phi field. \n");
-	fprintf(stderr, "Adding it in Grid_ReadGrid.C \n");
-	char *PhiName = "Phi";
-	PhiNum = NumberOfBaryonFields;
-	int PhiToAdd = PhiField;
-	this->AddFields(&PhiToAdd, 1);
-	DataLabel[PhiNum] = PhiName;
-      } else { 
-	if (0) 
-	  for (int n = 0; n < size; n++)
-	    BaryonField[PhiNum][n] = 0.;
-      }
-
-      /* if we restart from a different solvers output without a Phi_pField 
-	 and yet want to use the divergence cleaning, create here and set to zero */
-      if (UsePoissonDivergenceCleaning) {
-	int Phi_pNum; 
-	if ((Phi_pNum = FindField(Phi_pField, FieldType, NumberOfBaryonFields)) < 0) {
-	  fprintf(stderr, "Want to use divergence cleaning with no Phi_p field. \n");
-	  fprintf(stderr, "Adding it in Grid_ReadGrid.C \n");
-	  char *Phi_pName = "Phi_p";
-	  Phi_pNum = NumberOfBaryonFields;
-	  int Phi_pToAdd = Phi_pField;
-	  this->AddFields(&Phi_pToAdd, 1);
-	  DataLabel[Phi_pNum] = Phi_pName;
-	}
-      }
-
-      
-    } /* if HydroMethod == MHD */
-
-
     delete [] temp;
  
   }  // end:   if (NumberOfBaryonFields > 0 && ReadData &&
   //      (MyProcessorNumber == ProcessorNumber)) {
 
+  if (HydroMethod == MHD_RK) { // This is the MHD with Dedner divergence cleaning that needs an extra field
```

Comment:
Did this block of code move in a critical way?

Reply:
Yes, it did. See this [commit message](https://github.com/enzo-project/enzo-dev/pull/151/commits/34dfe0f1523149277003061d28dc11e98689458e). Before it was in the `ProcessorNumber == MyProcessorNumber` if-block, so it was only added on a single MPI process. This only worked in serial restarts before and crashed in parallel restarts. The `AddField()` routine adds the new field to the `FieldList` and only allocates and initializes the array on the "owner" MPI process.